src/overlays/actors/ovl_Obj_Tokeidai/z_obj_tokeidai.c

```diff
@@ -50,87 +42,803 @@ const ActorInit Obj_Tokeidai_InitVars = {
+            this->opaDList = D_0600D388;
+            break;
+        case OBJ_TOKEIDAI_TYPE_TOWER_WALLS_TERMINA_FIELD:
+            Actor_SetScale(&this->actor, 1.0f);
+            this->opaDList = D_06009A08;
+            this->actionFunc = ObjTokeidai_Walls_Idle;
+            break;
+        case OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_TERMINA_FIELD:
+            Actor_SetScale(&this->actor, 0.15f);
+            ObjTokeidai_TowerClock_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_CLOCK_TOWN:
+            ObjTokeidai_TowerClock_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_WALL_CLOCK:
+            Actor_SetScale(&this->actor, 0.02f);
+            this->actor.draw = ObjTokeidai_Clock_Draw;
+            ObjTokeidai_Clock_Init(this);
+            this->actionFunc = ObjTokeidai_WallClock_Idle;
+            break;
+        case OBJ_TOKEIDAI_TYPE_SMALL_WALL_CLOCK:
+            Actor_SetScale(&this->actor, 0.01f);
+            this->actor.draw = ObjTokeidai_Clock_Draw;
+            ObjTokeidai_Clock_Init(this);
+            this->actionFunc = ObjTokeidai_WallClock_Idle;
+            break;
+        case OBJ_TOKEIDAI_TYPE_COUNTERWEIGHT_TERMINA_FIELD:
+            Actor_SetScale(&this->actor, 0.15f);
+            ObjTokeidai_Counterweight_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_COUNTERWEIGHT_CLOCK_TOWN:
+            ObjTokeidai_Counterweight_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_STAIRCASE_INTO_TOWER:
+            this->opaDList = D_0600D8E8;
+            this->xluDList = D_0600D8E0;
+            this->actionFunc = ObjTokeidai_StaircaseIntoTower_Idle;
+            break;
+    }
+}
+
+void ObjTokeidai_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void ObjTokeidai_RotateOnMinuteChange(ObjTokeidai* this, s32 playSound) {
+    s32 currentMinute;
+
+    currentMinute = OBJ_TOKEIDAI_GET_CURRENT_MINUTE(this);
+    if (currentMinute != this->clockMinute) {
+        if (this->outerRingOrGearRotationTimer == 8 && playSound) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CLOCK_TOWER_SECOND_HAND);
+        }
+
+        if (this->outerRingOrGearRotationTimer >= 9) {
+            // This actually performs the rotation to the next minute
+            // for the outer ring or gear.
+            this->outerRingOrGearRotationalVelocity += 0x3C;
+            this->outerRingOrGearRotation += this->outerRingOrGearRotationalVelocity;
+        } else {
+            // This makes the outer ring or gear wiggle in place for a bit
+            // before rotating to the next position.
+            if ((this->outerRingOrGearRotationTimer & 3) == 0) {
+                this->outerRingOrGearRotation += 0x5A;
+            }
+            if ((this->outerRingOrGearRotationTimer & 3) == 1) {
+                this->outerRingOrGearRotation -= 0x5A;
+            }
+        }
+
+        this->outerRingOrGearRotationTimer++;
```

Comment:
```suggestion
        this->outerRingOrGearRotationCounter++;
```
If it increments up, would it be a counter instead of a timer?

Reply:
I vaguely remember having a similar discussion about this on Elforg, and just as with that actor, I prefer "timer" here for the following reasons:
1. It counts up, but the unit it is counting is frames, which is time. If I was going to call it a "counter", I would name it `outerRingOrGearRotationFrameCounter`, which is a little unwieldy. If I drop the word "frame", then you can't really tell it's actually measuring time, which is something that using "timer" avoids.
2. There are quite a few instances in the codebase of someone doing `timer++` or something similar. It's a fairly common pattern. Of course, I fully admit this is self-reinforcing; by calling this a "timer", I'm making the pattern even more common.